src/custom/state/application/atom.ts

```diff
@@ -0,0 +1,29 @@
+import { atom } from 'jotai'
+import { atomWithStorage, useUpdateAtom, useAtomValue } from 'jotai/utils'
+
+export { useUpdateAtom, useAtomValue }
+
+type FollowPendingTxPopup = {
+  showPopup: boolean
+  hideFollowTxPopup: boolean
+}
+
+/**
+ * Base atom that store the popup state that indicate how to follow a pending tx
+ */
+export const followPendingTxPopupAtom = atomWithStorage<FollowPendingTxPopup>('followPendingTxPopup', {
+  hideFollowTxPopup: false,
+  showPopup: false,
+})
+
+export const handleFollowPendingTxPopupAtom = atom(null, (_get, set, update: boolean) => {
+  set(followPendingTxPopupAtom, (prev) => ({ ...prev, showPopup: update }))
+})
+
+export const handleHidePopupPermanentlyAtom = atom(null, (_get, set, update: boolean) => {
+  set(followPendingTxPopupAtom, (prev) => ({ ...prev, hideFollowTxPopup: update }))
+})
+
+export const showFollowTxPopupAtom = atom(
+  (get) => get(followPendingTxPopupAtom).showPopup === true && !get(followPendingTxPopupAtom).hideFollowTxPopup
```

Comment:
It's probably cheap fetching it twice, but I'll sleep better at night having it called only once 😅 
Also, since the value will always be a boolean, you don't need to compare against `true`

```suggestion
  (get) => {
    const {showPopup, hideFollowTxPopup} = get(followPendingTxPopupAtom)
    return showPopup && !hideFollowTxPopup
  }
```

Reply:
Also, there's a util function to make a selector: https://jotai.org/docs/utils/select-atom

Which would make it even shorter :)